coin/coin.go

```diff
@@ -303,3 +306,72 @@ func (c Coin) normalize() (Coin, error) {
 	}
 	return c, nil
 }
+
+func (c *Coin) UnmarshalJSON(raw []byte) error {
+	// Prioritize human readable format that is a string in format
+	// "<whole>[.<fractional>] <ticker>"
+	var human string
+	if err := json.Unmarshal(raw, &human); err == nil {
+		whole, fract, ticker, err := parseHumanFormat(human)
+		if err == nil {
+			c.Whole = whole
+			c.Fractional = fract
+			c.Ticker = ticker
+		}
+		return err
+	}
+
+	// Fallback into the default unmarhaling. Because UnmarshalJSON method
+	// is provided, we can no longer use Coin type for this.
+	var coin struct {
+		Whole      int64
```

Comment:
I'm surprised you don't have to add `json:"whole"` here, so my experience is it would default to `{"Whole": 1...}`. But it seems it parsed in a test... If not an issue to add those explicit lower-case json flags, I'd like that

Reply:
>To unmarshal JSON into a struct, Unmarshal matches incoming object keys to the keys used by Marshal (either the struct field name or its tag), preferring an exact match but also accepting a case-insensitive match.  

It is even "better", guess how the below would parse?
```go
var x struct {
  Number int64
  NuMbEr int64
}
json.Unmarshal([]byte(`{"number": 123}`), &x)
```
:see_no_evil: